client/components/Main.jsx

```diff
@@ -99,10 +104,22 @@ export default class Main extends React.Component {
     this.socket.emit('pipelines:get');
   }
 
+  updateDisabledPipelinesOnSave(settings) {
+    const regexFilter = new RegExp(settings.filterRegexProps.value);
+    settings.disabledPipelines = this.state.pipelineNames.filter((p) => {
+      if (regexFilter.test(p)){
+        return !settings.filterRegexProps.active;
+      }
+      return settings.filterRegexProps.active;
+    });
+    this.saveSettings(settings);
+  }
+
   saveSettings(settings) {
     this.socket.emit('settings:update', {
       sortOrder: settings.sortOrder.name,
-      disabledPipelines: settings.disabledPipelines
+      disabledPipelines: settings.disabledPipelines,
+      filterRegexProps: settings.filterRegexProps
```

Comment:
Since you haven't done anything with the backend code. Does this works "out of the box"? Will this be stored in db?